build.rs

```diff
@@ -1,25 +1,49 @@
 use std::env;
 use std::fs;
-use std::path::PathBuf;
 
-#[cfg(target_os = "windows")]
-pub const LIB_FILE_NAME: &str = "bass.lib";
 pub const DLL_FILE_NAME: &str = "bass.dll";
 
-#[cfg(not(target_os = "windows"))]
-pub const LIB_FILE_NAME: &str = "libbass.so";
+pub const BINDINGS_FILE_NAME: &str = "bindings.rs";
 
-#[cfg(target_os = "windows")]
-pub const PREBUILT_BINDINGS_FILE_NAME: &str = "bindings_win.rs";
+/// Get the filename for the prebuilt bindings for our target platform.
+fn prebuilt_lib_binary_filename() -> &'static str {
+    // Note: We can't use cfg(target_os) because this breaks when cross-compiling.
+    let target_os = env::var("CARGO_CFG_TARGET_OS");
 
-#[cfg(not(target_os = "windows"))]
-pub const PREBUILT_BINDINGS_FILE_NAME: &str = "bindings_lnx.rs";
+    match target_os.as_ref().map(|x| &**x) {
+        Ok("linux") => "bass.lib",
+        Ok("windows") => "libbass.so",
+        Ok("macos") => "libbass.dylib",
+        Ok(unsupported_os) => panic!(
+            "Unsupported target os: \"{}\". Not sure what to link against, so we'll do nothing.",
+            unsupported_os
+        ),
+        Err(err) => panic!("Error reading target os for build: {}", err),
+    }
+}
 
-pub const BINDINGS_FILE_NAME: &str = "bindings.rs";
+/// Get the filename for the prebuilt bindings for our target platform.
+#[cfg(not(feature = "gen-bindings"))]
+fn prebuilt_bindings_filename() -> &'static str {
+    // Note: We can't use cfg(target_os) because this breaks when cross-compiling.
+    let target_os = env::var("CARGO_CFG_TARGET_OS");
+
+    match target_os.as_ref().map(|x| &**x) {
+        Ok("linux") => "bindings_lnx.rs",
+        Ok("windows") => "bindings_win.rs",
+        Ok(unsupported_os) => panic!(
```

Comment:
Can you upload the generated bindings for mac please? Ideally I don't want to require the bindgen feature by default.